scripts/publish.sh

```diff
@@ -239,6 +242,9 @@ publish_ecr() {
 	do
 		push_image_ecr ${AWS_ACCOUNT}.dkr.ecr.${AWS_REGION}.amazonaws.com/amazon/aws-for-fluent-bit-test:"$arch" \
 			${account_id}.dkr.ecr.${region}.amazonaws.com/aws-for-fluent-bit:"$arch"-${AWS_FOR_FLUENT_BIT_VERSION}
+		
+		# Added verification for sanity check even though we are pushing the same image that we scanned in build stage.
+		verify_ecr_image_scan ${region} aws-for-fluent-bit "$arch"-${AWS_FOR_FLUENT_BIT_VERSION} 
 	done
```

Comment:
Let's not do this. I think we should only scan in one part of the build, no where else. 

Otherwise I think there's a small chance we could get caught in a weird situation. Imagine in build the image scan passes, but then a new vulnerability is added to the database as the pipeline moves on, and then suddenly we've published to some regions but not others.

Reply:
If I understand correctly, we build the image and scan it to make sure there are no vulnerabilities. Then we push this same image to various regions and make the images public. So where would the new vulnerability be added in the pipeline, I guess I am missing something. Can you explain? Also is it possible that the same image in one region could be good but in another region bad? 

As I mentioned in the comments, it is not necessary for us to do this check as we have already scanned the image but I felt it was a good sanity check just before we make the image public. However as you mentioned, if a vulnerability creeps in, I think it is a good idea to have this in place so that we catch it early. 

Additionally, our goal is to perform image scan more frequently (1 scan per day). At this point, we will not build a new image, but scan the existing ones. So the issue you mentioned where having some regions with bad images could occur. How do we deal with that?

Reply:
> So where would the new vulnerability be added in the pipeline, I guess I am missing something. 

I think you're thinking of the vulnerability as something which only exists in the image. Which is technically true. But the vulnerability is only discovered if it exists in both the image and in the database of vulnerabilities, which is always being updated. 

So its possible you could have an image with a vulnerability that is unknown at build time, but that becomes "known" in the hour or so it takes for the pipeline and sync tasks to publish everywhere. Which then means that we only publish to some regions, which makes the rollback and messaging to customers complicated.

Reply:
I see now how it can be a problem. Thanks for this.